### PR TITLE
Add ClickHouse smoke tests for Altinity

### PR DESCRIPTION
**Pull request by ebyhr**
https://hub.docker.com/r/altinity/clickhouse-server/tags

### CODE REVIEW

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/AltinityQueryRunner.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.trino.plugin.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.trino.testing.QueryAssertions.copyTpchTables;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+public final class AltinityQueryRunner
```
Comment (findepi) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/AltinityQueryRunner.java:32]: We named `XxxQueryRunner` classes after connectors, so it's still `ClickHouse...`

call it ClickHouseAltinityQueryRunner

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/AltinityQueryRunner.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+{
+    public static final String TPCH_SCHEMA = "tpch";
+
+    private AltinityQueryRunner() {}
+
+    public static DistributedQueryRunner createAltinityQueryRunner(
+            TestingAltinityServer server,
```
Comment (findepi) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/AltinityQueryRunner.java:39]: This is probably the only difference between ClickHouseQueryRunner and this class.
We don't need that duplication, as all we need is a) jdbc url and b) ability to create tpch schema.

BTW Can `server.execute("CREATE DATABASE tpch"`) can be done via the connector now?

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestingAltinityServer.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import java.sql.Statement;
+
+import static java.lang.String.format;
+import static org.testcontainers.containers.ClickHouseContainer.HTTP_PORT;
+import static org.testcontainers.utility.MountableFile.forClasspathResource;
+
+/**
+ * @see <a href="https://docs.altinity.com/altinitystablebuilds/">Altinity Stable Builds</a>
+ */
+public class TestingAltinityServer
```
Comment (findepi) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestingAltinityServer.java:31]: What's the difference between this class and TestingClickHouseServer?

Is it only in docker image name/versions being used?
Why not reuse `TestingClickHouseServer`?

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/AltinityQueryRunner.java (hunk 2 above)
Comment (ebyhr) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/AltinityQueryRunner.java]: Changed to `queryRunner.execute("CREATE SCHEMA " + TPCH_SCHEMA);`

**Comment by findepi** [comment]
@hashhar ptal

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestAltinityConnectorSmokeTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createClickHouseQueryRunner(
+                closeAfterClass(new TestingClickHouseServer(ALTINITY_DEFAULT_IMAGE)),
+                ImmutableMap.of(),
+                ImmutableMap.<String, String>builder()
+                        .put("clickhouse.map-string-as-varchar", "true")
```
Comment (hashhar) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestAltinityConnectorSmokeTest.java:34]: Do we remember why this config is added to query runners by default?

Let's add a comment.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestAltinityConnectorSmokeTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                        .buildOrThrow(),
+                REQUIRED_TPCH_TABLES);
+    }
+
+    @Override
+    public void testRenameSchema()
+    {
+        // Override because RENAME DATABASE statement isn't supported in version < v20.7.2.30-stable
```
Comment (hashhar) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestAltinityConnectorSmokeTest.java:42]: This version number seems different from the one already mentioned in Clickhouse tests. Was this intended?

Comment (ebyhr) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestAltinityConnectorSmokeTest.java:42]: The comment in ClickHouse mentions the default engine, but this comment mentions syntax.

**Review by hashhar — APPROVED** [review]
LGTM % ci

Thanks for the initial review @findepi.

**Pull request merged**